Library/Homebrew/test/livecheck/strategy/sparkle_spec.rb

```diff
@@ -147,29 +285,51 @@
     end
   end
 
+  # The Sparkle Strategy only sorts items in `#versions_from_content`.
+  # This block thus expects things to remain in the order they're listed
+  # in the `xml` and `items` data structures.
   describe "::items_from_content" do
     let(:items_from_appcast) { sparkle.items_from_content(xml[:appcast]) }
-    let(:first_item) { items_from_appcast[0] }
 
     it "returns nil if content is blank" do
       expect(sparkle.items_from_content("")).to eq([])
     end
 
     it "returns an array of Items when given XML data" do
       expect(items_from_appcast).to eq(items[:appcast])
-      expect(first_item.title).to eq(item_hash[0][:title])
-      expect(first_item.pub_date).to eq(Time.parse(item_hash[0][:pub_date]))
-      expect(first_item.url).to eq(item_hash[0][:url])
-      expect(first_item.short_version).to eq(item_hash[0][:short_version])
-      expect(first_item.version).to eq(item_hash[0][:version])
+      expect(items_from_appcast[0].title).to eq(item_hashes[:version_123][:title])
+      expect(items_from_appcast[0].pub_date).to eq(Time.parse(item_hashes[:version_123][:pub_date]))
+      expect(items_from_appcast[0].url).to eq(item_hashes[:version_123][:url])
+      expect(items_from_appcast[0].short_version).to eq(item_hashes[:version_123][:short_version])
+      expect(items_from_appcast[0].version).to eq(item_hashes[:version_123][:version])
 
       expect(sparkle.items_from_content(xml[:beta_channel_item])).to eq(items[:beta_channel_item])
       expect(sparkle.items_from_content(xml[:no_versions_item])).to eq(items[:no_versions_item])
     end
   end
 
+  # The Sparkle Strategy _sorts_ items in `#versions_from_content`. So
+  # this block also has to sort the data in `xml` and `items` in order to
+  # expect the right results.
   describe "::versions_from_content" do
-    let(:subbed_items) { items[:appcast].map { |item| item.nice_version.sub("1", "0") } }
+    # We're `sub`ing here because we want to be able to process a block in
+    # `#versions_from_content` and this gives us a way to verify that we
+    # can.
+    let(:subbed_versions_in_sort_order) {
+      items_in_version_sort_order
+        .map { |item| item.nice_version.sub("1", "0") }
+    }
```

Comment:
```suggestion
    let(:subbed_items) { items[:appcast_sorted].map { |item| item.nice_version.sub("1", "0") } }
```

With the addition of the `:appcast_sorted` array, this can return to the previous formatting. Past that, I don't think the additional commentary here is necessary/beneficial (i.e., one can simply look a few lines down to see how `subbed_items` is used).